code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -24,76 +77,156 @@
-
+	addtimer(CALLBACK(src, .proc/sneak_attack_cooldown), HUNTER_POUNCE_SNEAKATTACK_DELAY) //Short delay before we can sneak attack.
 
-/mob/living/carbon/xenomorph/hunter/proc/cancel_stealth() //This happens if we take damage, attack, pounce, toggle stealth off, and do other such exciting stealth breaking activities.
+/datum/action/xeno_action/stealth/proc/cancel_stealth() //This happens if we take damage, attack, pounce, toggle stealth off, and do other such exciting stealth breaking activities.
 	if(!stealth)//sanity check/safeguard
 		return
-	to_chat(src, "<span class='xenodanger'>We emerge from the shadows.</span>")
-	UnregisterSignal(src, COMSIG_MOVABLE_MOVED) //This should be handled on the ability datum or a component.
+	to_chat(owner, "<span class='xenodanger'>We emerge from the shadows.</span>")
+	UnregisterSignal(owner, COMSIG_MOVABLE_MOVED) //This should be handled on the ability datum or a component.
 	stealth = FALSE
 	can_sneak_attack = FALSE
-	alpha = 255 //no transparency/translucency
+	owner.alpha = 255 //no transparency/translucency
 
-// ***************************************
-// *********** Pounce/sneak attack
-// ***************************************
-/datum/action/xeno_action/activable/pounce/hunter
-	plasma_cost = 20
-	range = 7
+/datum/action/xeno_action/stealth/proc/sneak_attack_cooldown()
+	if(!stealth || can_sneak_attack)
+		return
+	can_sneak_attack = TRUE
+	to_chat(owner, "<span class='xenodanger'>We're ready to use Sneak Attack while stealthed.</span>")
+	playsound(owner, "sound/effects/xeno_newlarva.ogg", 25, 0, 1)
 
-/datum/action/xeno_action/activable/pounce/hunter/prepare_to_pounce()
-	. = ..()
+/datum/action/xeno_action/stealth/proc/handle_stealth()
+	if(!stealth)
+		return
+	var/mob/living/livingowner = owner
+	if(owner.stat != CONSCIOUS || owner.lying_angle || livingowner.resting) //Can't stealth while unconscious/resting
+		cancel_stealth()
+		return
+	//Initial stealth
+	if(last_stealth > world.time - HUNTER_STEALTH_INITIAL_DELAY) //We don't start out at max invisibility
+		owner.alpha = HUNTER_STEALTH_RUN_ALPHA * stealth_alpha_multiplier
+		return
+	//Stationary stealth
+	else if(owner.last_move_intent < world.time - HUNTER_STEALTH_STEALTH_DELAY) //If we're standing still for 4 seconds we become almost completely invisible
+		owner.alpha = HUNTER_STEALTH_STILL_ALPHA * stealth_alpha_multiplier
+	//Walking stealth
+	else if(owner.m_intent == MOVE_INTENT_WALK)
+		owner.alpha = HUNTER_STEALTH_WALK_ALPHA * stealth_alpha_multiplier
+	//Running stealth
+	else
+		owner.alpha = HUNTER_STEALTH_RUN_ALPHA * stealth_alpha_multiplier
+	//If we have 0 plasma after expending stealth's upkeep plasma, end stealth.
+	var/mob/living/carbon/xenomorph/xenoowner = owner
+	if(!xenoowner.plasma_stored)
+		to_chat(xenoowner, "<span class='xenodanger'>We lack sufficient plasma to remain camouflaged.</span>")
+		cancel_stealth()

```

Comment:
This check should be at the beginning, no?

Reply:
It's how it worked before, this code is 1:1 the old code functionality